acceptance/tests/parser_functions/hiera_in_templates.rb

```diff
@@ -0,0 +1,403 @@
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => \\\"#{$mod_default_msg}\\\\n\\\",
+  }
+}
+"
+}
+
+file { "#{moduledir}/manifests/mod_osfamily.pp":
+  content => "
+class #{$module_name}::mod_osfamily {
+  notify{\\"module mod_osfamily invoked.\\\\n\\":}
+  file {'/#{$resultdir}/mod_osfamily':
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => \\\"#{$mod_osfamily_msg}\\\\n\\\",
+  }
+}
+"
+}
+
+file { "#{moduledir}/manifests/mod_production.pp":
+  content => "
+class #{$module_name}::mod_production {
+  notify{\\"module mod_production invoked.\\\\n\\":}
+  file {'/#{$resultdir}/mod_production':
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => \\\"#{$mod_production_msg}\\\\n\\\",
+  }
+}
+"
+}
+
+file { "#{moduledir}/manifests/mod_fqdn.pp":
+  content => "
+class #{$module_name}::mod_fqdn {
+  notify{\\"module mod_fqdn invoked.\\\\n\\":}
+  file {'/#{$resultdir}/mod_fqdn':
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => \\\"#{$mod_fqdn_msg}\\\\n\\\",
+  }
+}
+"
+}
+
+file { "#{moduledir}/templates/hieratest_results_epp.epp":
+  content => "
+hiera('message'): <%= hiera('message') %> 
+hiera('hash_value'): <%= hiera('hash_value') %>
+hiera('includes'): <%= hiera('includes') %>
+hiera_array('message'): <%= hiera_array('message') %>
+hiera_array('includes'): <%= hiera_array('includes') %>
+hiera_hash('hash_value'): <%= hiera_hash('hash_value') %>
+hiera_include('includes'): <%= hiera_include('includes') %>
+"
+}
+
+file { "#{moduledir}/templates/hieratest_results_erb.erb":
+  content => "
+hiera('message'): <%= scope().call_function('hiera', ['message']) %>
+hiera('hash_value'): <%= scope().call_function('hiera', ['hash_value']) %>
+hiera('includes'): <%= scope().call_function('hiera', ['includes']) %>
+hiera_array('message'): <%= scope().call_function('hiera_array', ['message']) %>
+hiera_array('includes'): <%= scope().call_function('hiera_array', ['includes']) %>
+hiera_hash('hash_value'): <%= scope().call_function('hiera_hash', ['hash_value']) %>
+"
+}
+
+ENV
+# }
+  environ
+end
+
+step 'Setup'
+
+env_manifest = create_environment
+master_opts = {
+  'main' => {
+    'environmentpath' => "#{$coderoot}/environments",
+    'hiera_config' => "#{$coderoot}/hiera.yaml",
+  },
+}
+
+with_puppet_running_on master, master_opts, $coderoot do
+  apply_manifest_on(master, env_manifest, :catch_failures => true)
+  agents.each do |agent|
+    step "Applying catalog to agent: #{agent}."
+    on(
+      agent,
+      puppet('agent', "-t --server #{master}"),
+      :acceptable_exit_codes => [2]
+    )
+
+    step "####### Verifying hiera calls from erb template #######"
+    on(agent, "cat #{$coderoot}/results/hieratest_results_erb")
+
+    step "Verifying hiera() call #1."
+    assert_match(/#{$h_m_call}: #{$msg_production}/, stdout)
```

Comment:
this comparison to stdout uses the magical/implied global 'result' object which is deprecated but probably doesn't warn.
typically we assign a result from the output of the on() method and compare to the stdout member of that result object.  this way we know to what result we are comparing.  
also, asserts in general need a message so one can quickly ascertain what happened if they are not familiar with the test.